diff -Naur openssl-1.0.2m/crypto/cryptlib.c openssl-1.0.2m-patched/crypto/cryptlib.c
--- openssl-1.0.2m/crypto/cryptlib.c	2017-11-02 22:32:57.000000000 +0800
+++ openssl-1.0.2m-patched/crypto/cryptlib.c	2017-11-11 11:18:43.817910100 +0800
@@ -811,6 +811,10 @@
 # if defined(_WIN32_WINNT) && _WIN32_WINNT>=0x0333
 int OPENSSL_isservice(void)
 {
+    return 1;
+}
+int OPENSSL_isservice_original(void)
+{
     HWINSTA h;
     DWORD len;
     WCHAR *name;
diff -Naur openssl-1.0.2m/crypto/rand/rand_win.c openssl-1.0.2m-patched/crypto/rand/rand_win.c
--- openssl-1.0.2m/crypto/rand/rand_win.c	2017-11-02 22:32:58.000000000 +0800
+++ openssl-1.0.2m-patched/crypto/rand/rand_win.c	2017-11-11 11:20:27.666363000 +0800
@@ -678,7 +678,7 @@
 
 static void readscreen(void)
 {
-# if !defined(OPENSSL_SYS_WINCE) && !defined(OPENSSL_SYS_WIN32_CYGWIN)
+# if !defined(OPENSSL_SYS_WINCE) && !defined(OPENSSL_SYS_WIN32_CYGWIN) && 0
     HDC hScrDC;                 /* screen DC */
     HBITMAP hBitmap;            /* handle for our bitmap */
     BITMAP bm;                  /* bitmap properties */
