diff -Naur openssl-1.0.2m/util/mk1mf.pl openssl-1.0.2m-patched/util/mk1mf.pl
--- openssl-1.0.2m/util/mk1mf.pl	2017-11-02 22:32:58.000000000 +0800
+++ openssl-1.0.2m-patched/util/mk1mf.pl	2017-11-11 10:40:51.097409700 +0800
@@ -523,7 +523,7 @@
 # You should not need to touch anything below this point
 ######################################################
 
-E_EXE=openssl
+E_EXE=openssl$libsuffix
 SSL=$ssl
 CRYPTO=$crypto
 
@@ -1236,6 +1236,8 @@
 		"shlib" => \$shlib,
 		"dll" => \$shlib,
 		"shared" => 0,
+		"debug_lib" => \$debug_lib,
+		"static_lib" => \$static_lib,
 		"no-sctp" => 0,
 		"no-srtp" => 0,
 		"no-gmp" => 0,
diff -Naur openssl-1.0.2m/util/mkdef.pl openssl-1.0.2m-patched/util/mkdef.pl
--- openssl-1.0.2m/util/mkdef.pl	2017-11-02 22:32:58.000000000 +0800
+++ openssl-1.0.2m-patched/util/mkdef.pl	2017-11-11 10:38:30.896605000 +0800
@@ -55,6 +55,8 @@
 #
 
 my $debug=0;
+my $debug_lib=0;
+my $static_lib=0;
 
 my $crypto_num= "util/libeay.num";
 my $ssl_num=    "util/ssleay.num";
@@ -157,6 +159,8 @@
 foreach (@ARGV, split(/ /, $options))
 	{
 	$debug=1 if $_ eq "debug";
+	$debug_lib=1 if $_ eq "debug_lib";
+	$static_lib=1 if $_ eq "static_lib";
 	$W32=1 if $_ eq "32";
 	$W16=1 if $_ eq "16";
 	if($_ eq "NT") {
@@ -1322,7 +1326,20 @@
 		  # Vendor field can't contain colon, drat; so we omit http://
 		  $description = "\@#$http_vendor:$version#\@$what; DLL for library $name.  Build for EMX -Zmtd";
 		}
-
+		
+	$libsuffix="";
+	if ($static_lib) {
+		$libsuffix="MT";
+	} else {
+		$libsuffix="MD";
+	}
+	
+	if ($debug_lib) {
+		$libsuffix=$libsuffix."d";
+	}
+		
+	$libname=$libname.$libsuffix;
+	
 	print OUT <<"EOF";
 ;
 ; Definition file for the DLL version of the $name library from OpenSSL
diff -Naur openssl-1.0.2m/util/pl/VC-32.pl openssl-1.0.2m-patched/util/pl/VC-32.pl
--- openssl-1.0.2m/util/pl/VC-32.pl	2017-11-02 22:32:59.000000000 +0800
+++ openssl-1.0.2m-patched/util/pl/VC-32.pl	2017-11-11 10:37:20.442798100 +0800
@@ -15,6 +15,20 @@
 	{
 	$crypto="libeay32";
 	}
+	
+$libsuffix="";
+if ($static_lib) {
+	$libsuffix="MT";
+} else {
+	$libsuffix="MD";
+}
+
+if ($debug_lib) {
+	$libsuffix=$libsuffix."d";
+}
+	
+$ssl=$ssl.$libsuffix;
+$crypto=$crypto.$libsuffix;
 
 $o='\\';
 $cp='$(PERL) util/copy.pl';
@@ -155,8 +169,8 @@
 	}
 
 # generate symbols.pdb unconditionally
-$app_cflag.=" /Zi /Fd\$(TMP_D)/app";
-$lib_cflag.=" /Zi /Fd\$(TMP_D)/lib";
+$app_cflag.=" /Zi /Fd\$(TMP_D)/app".$libsuffix;
+$lib_cflag.=" /Zi /Fd\$(TMP_D)/lib".$libsuffix;
 $lflags.=" /debug";
 
 $obj='.obj';
